File: MyStagePass.Services/Database/City.cs
namespace MyStagePass.Services.Database
{
	public class City
	{
		public int CityID { get; set; }
		public string? Name { get; set; }
		public int CountryID {  get; set; }
		public virtual Country Country { get; set; } = null!;
		public virtual ICollection<Location> Locations { get; set; } = new List<Location>();
	}
}

File: UI/mystagepass_admin/lib/models/City/city.dart
import 'package:json_annotation/json_annotation.dart';
import '../Country/country.dart';
import '../Location/location.dart';
part 'city.g.dart';

@JsonSerializable()
class City {
  @JsonKey(name: 'cityID')
  int? cityId;
  String? name;
  bool? isActive;
  @JsonKey(name: 'countryID')
  int? countryId;
  List<Location>? locations;

  Country? country;

  City({
    this.cityId,
    this.name,
    this.isActive,
    this.countryId,
    this.locations,
    this.country,
  });

  factory City.fromJson(Map<String, dynamic> json) => _$CityFromJson(json);
  Map<String, dynamic> toJson() => _$CityToJson(this);
}
